src/rpc/handlers/AMMInfo.cpp

```diff
@@ -0,0 +1,225 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2023, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <data/DBHelpers.h>
+#include <rpc/AMMHelpers.h>
+#include <rpc/handlers/AMMInfo.h>
+
+#include <ripple/protocol/AMMCore.h>
+
+namespace {
+
+std::string
+toIso8601(ripple::NetClock::time_point tp)
+{
+    using namespace std::chrono;
+    static auto constexpr rippleEpochOffset = seconds{rippleEpochStart};
+
+    return date::format(
+        "%Y-%Om-%dT%H:%M:%OS%z",
+        date::sys_time<system_clock::duration>(system_clock::time_point{tp.time_since_epoch() + rippleEpochOffset})
+    );
+};
+
+}  // namespace
+
+namespace rpc {
+AMMInfoHandler::Result
+AMMInfoHandler::process(AMMInfoHandler::Input input, Context const& ctx) const
+{
+    using namespace ripple;
+
+    if (!input.ammAccount && (input.issue1 == ripple::noIssue() || input.issue2 == ripple::noIssue()))
+        return Error{Status{RippledError::rpcINVALID_PARAMS}};
+
+    auto const range = sharedPtrBackend_->fetchLedgerRange();
+    auto const lgrInfoOrStatus = getLedgerInfoFromHashOrSeq(
+        *sharedPtrBackend_, ctx.yield, input.ledgerHash, input.ledgerIndex, range->maxSequence
+    );
+
+    if (auto const status = std::get_if<Status>(&lgrInfoOrStatus))
+        return Error{*status};
+
+    auto const lgrInfo = std::get<LedgerInfo>(lgrInfoOrStatus);
+    ripple::uint256 ammID;
+    if (input.ammAccount) {
+        auto accountKeylet = ripple::keylet::account(*input.ammAccount);
```

Comment:
we need to check the validity of the account , make sure 
1 it is a valid account
2 it is a valid amm account
otherwise the Clio may crash